### PR TITLE
refine image classification train script

### PR DESCRIPTION
**Pull request by shippingwang**
- refine Image classification code

1. refine set model category func.
2. remove exp decay, will rewrite it in the next commit.
3. print learning rate when running the iteration.

- [ ] add profile file
- [ ] add more efficient reader

### CODE REVIEW

File: fluid/PaddleCV/image_classification/train.py [hunk 1]
```diff
@@ -19,8 +18,10 @@
 from utils.learning_rate import cosine_decay
 from utils.fp16_utils import create_master_params_grads, master_param_to_train_param
 from utility import add_arguments, print_arguments
-import models
-import models_name
+import paddle.fluid.profiler as profiler
```
Comment (qingqing01) [inline fluid/PaddleCV/image_classification/train.py]: Remove this line.

File: fluid/PaddleCV/image_classification/train.py [hunk 2]
```diff
@@ -19,8 +18,10 @@
...
+
+IMAGENET1000 = 1281167
+#IMAGENET100 = 128660
```
Comment (qingqing01) [inline fluid/PaddleCV/image_classification/train.py]: Do not change the origin code ?

File: fluid/PaddleCV/image_classification/train.py [hunk 3]
```diff
@@ -89,43 +98,43 @@ def optimizer_setting(params):
...
-        learning_decay_rate_factor=ls["learning_decay_rate_factor"]
-        num_epochs_per_decay = ls["num_epochs_per_decay"]
-        NUM_GPUS = 1
-
+        start_lr = 0.5
+        l2_decay = params["l2_decay"]
+        momentum_rate = params["momentum_rate"]
+        end_lr = 0
+        total_step = int((total_images / batch_size) * num_epochs)
+        print("linear decay total steps: ", total_step)
```
Comment (qingqing01) [inline fluid/PaddleCV/image_classification/train.py]: Remove this line.

File: fluid/PaddleCV/image_classification/train.py [hunk 4]
```diff
@@ -190,19 +200,25 @@ def build_program(is_train, main_prog, startup_prog, args):
...
                     optimizer.apply_gradients(master_params_grads)
-                    master_param_to_train_param(master_params_grads, params_grads, main_prog)
+                    master_param_to_train_param(master_params_grads,
+                                                params_grads, main_prog)
                 else:
                     optimizer.minimize(avg_cost)
+                global_lr = optimizer._global_learning_rate()
 
-    return py_reader, avg_cost, acc_top1, acc_top5
+    if is_train:
```
Comment (qingqing01) [inline fluid/PaddleCV/image_classification/train.py:215]: train.py里还有非train的逻辑吗？

Comment (shippingwang) [reply fluid/PaddleCV/image_classification/train.py:215]: test 也在里面

File: fluid/PaddleCV/image_classification/train.py (hunk 3 above)
Comment (shippingwang) [reply fluid/PaddleCV/image_classification/train.py]: done

File: fluid/PaddleCV/image_classification/train.py (hunk 1 above)
Comment (shippingwang) [reply fluid/PaddleCV/image_classification/train.py]: ok

**Review by qingqing01 — APPROVED** [review]
